src/services/chirpstack/chirpstack-gateway.service.ts

```diff
@@ -555,4 +583,89 @@ export class ChirpstackGatewayService extends GenericChirpstackConfigurationServ
 
     return orderBy;
   }
+
+  checkForMinMaxNumbers(dto: UpdateGatewayDto) {
+    if (dto.gateway.minimumPackages > dto.gateway.maximumPackages) {
+      throw new BadRequestException({
+        success: false,
+        error: "Minimum has to be under maximum, and maximum has to be over minimum",
+      });
+    }
+  }
+  async checkForAlarms(gateways: GatewayResponseDto[]) {
+    for (let index = 0; index < gateways.length; index++) {
+      if (gateways[index].notificationOffline) {
+        await this.checkForNotificationOfflineAlarms(gateways[index]);
+      }
+    }
+  }
+
+  private async checkForNotificationOfflineAlarms(gateway: GatewayResponseDto) {
+    const actualDate = dayjs();
+    const lastSeen = dayjs(gateway.lastSeenAt);
+    if (actualDate.diff(lastSeen, "minute") > gateway.amountOfMinutes && !gateway.hasSentOfflineNotification) {
+      await this.oS2IoTMail.sendMail({
+        to: gateway.alarmMail,
+        subject: `OS2iot alarm: ${gateway.name} er offline`,
+        html: `<p>OS2iot alarm</p>
+               <p>Gateway’en ${gateway.name} er offline.</p>
+               <p>Der udsendes først besked igen, når gateway’en kommer online.</p>
+               <p>Link: <a href="${this.configService.get<string>("frontend.baseurl")}/gateways/gateway-detail/${
+          gateway.gatewayId
+        }">${this.configService.get<string>("frontend.baseurl")}/gateways/gateway-detail/${gateway.gatewayId}</a></p>
+              `,
+      });
+      await this.gatewayRepository.update({ gatewayId: gateway.gatewayId }, { hasSentOfflineNotification: true });
+    } else if (gateway.hasSentOfflineNotification && actualDate.diff(lastSeen, "minute") <= gateway.amountOfMinutes) {
```

Comment:
Kinda edge casey, hvis ens amount of of minutes er lav så kan den i teorien kommer over diff og derved aldrig sende en online besked

Reply:
Det mindste tal som kan indtastes er 1, og man tager ikke decimal tal - bør det så stadig kunne lade sig gøre? Eftersom den tjekker hvert minut. Jeg kan evt sige Math.ceil på diff, så den runder op til 1, hvis det fx var 0.7 min siden - så bør den altid bliver ramt, eftersom amountOfMinutes mindst kan være 1?

Reply:
Hmm. Jeg har gjort så den mindst kan være 1. Så skal man virkelig være uheldig i mange minutter i træk, før at den ikke sender.. korrekt? Eller skal jeg gentænke det?

Reply:
Måske kan det egentlig bare fjernes..